src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ByteViewer.cs

```diff
@@ -258,12 +257,20 @@ private void DrawHex(Graphics g, byte[] lineBuffer, int line)
     /// </summary>
     private void DrawLines(Graphics g, int startLine, int linesCount)
     {
+        if (linesCount == 0)
+        {
+            return;
+        }
+
+        int maxLength = _columnCount * 3 + 1;
+        using BufferScope<char> charsBufferScope = new(maxLength);
```

Comment:
Another improvement you can do here is to provide a stack allocated space. Something like this provides a nice safe alternative:

```suggestion
        using BufferScope<char> charsBuffer = new(stackalloc char[256], maxLength);
```

You also don't need to assign it to a local. The pattern is to just use it and let the implicit cast operator get out the span.

Reply:
I assigned it to a local because I wanted to avoid having the implicit cast happening over and over inside the loop, but I can remove it if you think that's overkill

Reply:
Yeah, it isn't going to measure in the scope of the cost of rendering here, so I'd go with simpler.